public/pages/createDetector/containers/DataSource/DataSource.tsx

```diff
@@ -131,11 +155,25 @@ function DataSource(props: DataFilterProps) {
             isInvalid={isInvalid(field.name, form)}
             error={getError(field.name, form)}
           >
-            <EuiSelect
-              {...field}
+            <EuiComboBox
               id="timeField"
               placeholder="Find timestamp"
               options={timeStampFieldOptions}
+              onSearchChange={handleSearchChange}
+              onCreateOption={(createdOption: string) => {
+                const normalizedOptions = createdOption.trim();
+                if (!normalizedOptions) return;
+                form.setFieldValue('timeField', normalizedOptions);
+              }}
+              onBlur={() => {
+                form.setFieldTouched('timeField', true);
+              }}
+              onChange={options => {
+                form.setFieldValue('timeField', get(options, '0.label'));
+              }}
+              selectedOptions={(field.value && [{ label: field.value }]) || []}
+              singleSelection={true}
+              isClearable={false}
```

Comment:
Could we maybe change this to `isClearable={true}`? I've noticed it seems kind of annoying when trying to change the selected index in the combobox. The only way to do it is backspace while the cursor is in the box.